src/test/regress/input/temp_tablespaces.source

```diff
@@ -138,3 +189,249 @@ drop tablespace mytempsp1;
+DECLARE
+    owo int;
+    session_file_ts_name text;
+    temp_table_ts_name text;
+BEGIN
+    SET optimizer = off;
+    SET statement_mem = '1MB';
+
+    CREATE TEMPORARY TABLE tts_baz (a int) DISTRIBUTED BY (a);
+
+    -- See the tablespace where temporary table was created. If there is no
+    -- tablespace for this table, it's in default tablespace.
+    SELECT ts.spcname
+        FROM pg_tablespace ts
+        JOIN pg_class c ON ts.oid = c.reltablespace AND c.relname = 'tts_baz'
+        LIMIT 1
+        INTO temp_table_ts_name;
+
+    DROP TABLE tts_baz;
+
+    -- Find out the tablespace that will be used for files.
+    -- N = gp_session_id % number of tablespaces in GUC.
+    SELECT 'temp_files' || (current_setting('gp_session_id')::int % n_temp_spill_files_tablespaces)
+        INTO session_file_ts_name;
+
+    -- Create the temporary directory.
+    WITH out11 AS (
+        SELECT * FROM (
+            WITH cte(a,b) AS (SELECT a-1, b+1 FROM tts_foo)
+            SELECT * FROM (SELECT * FROM cte UNION ALL SELECT * FROM cte) AS X JOIN tts_bar ON b = c
+            UNION ALL SELECT *, 1, 2 FROM cte
+        ) AS XY
+        JOIN tts_jazz on c = e AND b = f
+    )
+    SELECT count(*) FROM out11 INTO owo;
+
+    -- Start monitoring temporary tablespace usage.
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), 'default', gp_temptablespace_path((SELECT dattablespace FROM pg_database WHERE datname = current_database() LIMIT 1)));
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), 'temp_tables', gp_temptablespace_path((SELECT oid FROM pg_tablespace WHERE spcname = 'temp_tables' LIMIT 1)));
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), session_file_ts_name, gp_temptablespace_path((SELECT oid FROM pg_tablespace WHERE spcname = session_file_ts_name LIMIT 1)));
+
+    -- Run the query again to create temporary hashjoin files in the tablespaces.
+    WITH out11 AS (
+        SELECT * FROM (
+            WITH cte(a,b) AS (SELECT a-1, b+1 FROM tts_foo)
+            SELECT * FROM (SELECT * FROM cte UNION ALL SELECT * FROM cte) AS X JOIN tts_bar ON b = c
+            UNION ALL SELECT *, 1, 2 FROM cte
+        ) AS XY
+        JOIN tts_jazz on c = e AND b = f
+    )
+    SELECT count(*) FROM out11 INTO owo;
+
+    PERFORM gp_tablespace_watch_stop();
+
+    -- Check the existence of temporary hashjoin files.
+    RETURN QUERY SELECT
+        coalesce(temp_table_ts_name, 'default'),
+        CASE
+            WHEN gp_tablespace_watch_match(dbid, 'default', 'HashJoin') > 0 THEN 'default'
+            WHEN gp_tablespace_watch_match(dbid, 'temp_tables', 'HashJoin') > 0 THEN 'temp_tables'
+            WHEN gp_tablespace_watch_match(dbid, session_file_ts_name, 'HashJoin') > 0 THEN 'temp_filesN'
+        END
+        FROM gp_segment_configuration
+        WHERE role = 'p' AND content >= 0;
+END;
+$$ LANGUAGE plpgsql;
+
+--
+-- Global GUC value tests:
+-- FIXME: We shouldn't be forced to use \c to update GUCs on segments.
```

Comment:
We fixed this issue earlier. Now `gpstop -u` (in ignore blocks below) shouldn't require running `\c` to reload the GUCs.

Reply:
What has to be changed here, just removing the commend and `\c`, or is there something else?

Reply:
Yeah, just remove `\c` in blocks below. The GUCs should be updated correctly now.